src/main/resources/avro/common.avdl

```diff
@@ -45,6 +53,37 @@ record Position {
 }
 
 /**
+A `Segment` is a substring of a `Reference`. One or more `Segment`s (possibly
+from different `Reference`s) can be strung together end to end to form a `Path`.
+*/
+record Segment {
+  /**
+  The `Position` at which this `Segment` starts. The `Position` also defines the
+  direction in which the segment continues.
+  */
+  Position start;
+
+  /**
+  The number of bases that are covered by this segment.
+  */
+  long length;
+}
+
+/**
+A `Path` is an ordered list of `Segment`s. In general any contiguous path
+through the sequence graph of `Reference`s, with no novel adjacencies, can be
+represented by a `Path`.
+*/
+record Path {
+  /**
+  Require that segments abut in order, changing between `Reference`s only where they actually connect.
+  Also require that segments are maximal, i.e. that
+  consecutive elements of the array cannot be replaced by a single element.
+  */
+  array<Segment> segments;
```

Comment:
Nit: might be a good idea to provide a default value, e.g. `array<Segment> segments = [];`